### PR TITLE
Ensures ip addresses for all hostnames are resolved, solves issue with docker hostnames

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
Sorry for sitting on this so long.
I think it is a better implementation than #71  and should solve #75 
@matthil @adriancole  @ajanques @ssteeg-mdsol  @ykitamura-mdsol

### CODE REVIEW

File: lib/zipkin-tracer/hostname_resolver.rb [hunk 1]
```diff
@@ -19,6 +21,13 @@ def spans_with_ips(spans)
     LOCALHOST_I32 = 0x7f000001.freeze
     IP_FIELD = 3
 
+    def resolved_ip_address?(ip_string)
+      # When the ip_format is string, we will match with one of these two regexp
+      # When the ip_format is :i32 (used by kafka), we just check the string is a number
+      !!((ip_string =~ Resolv::IPv4::Regex) || (ip_string =~ Resolv::IPv6::Regex)) ||
```
Comment (adriancole) [inline lib/zipkin-tracer/hostname_resolver.rb]: if we found an ipv6 stuffed in the ipv4 field, we should probably move it to the ipv6 one :)

Comment (jcarres-mdsol) [inline lib/zipkin-tracer/hostname_resolver.rb]: This is in preparation when we support ipv6. We will need a different pr for that

Comment (ykitamura-mdsol) [inline lib/zipkin-tracer/hostname_resolver.rb]: you can use`Regexp.union`, if you like
```ruby
ip_string =~ Regexp.union(Resolv::IPv4::Regex, Resolv::IPv6::Regex)
```

**Comment by ykitamura-mdsol** [comment]
Looks good to me

**Pull request merged**